packages/headless/src/features/attached-results/attached-results-actions.ts

```diff
@@ -0,0 +1,91 @@
+import {ArrayValue, RecordValue, SchemaValidationError} from '@coveo/bueno';
+import {createAction} from '@reduxjs/toolkit';
+import {
+  validatePayload,
+  requiredNonEmptyString,
+  nonEmptyString,
+  serializeSchemaValidationError,
+} from '../../utils/validate-payload';
+import {AttachedResult} from './attached-results-state';
+
+export interface SetAttachedResultsActionCreatorPayload {
+  /**
+   * The array of all attached results.
+   */
+  results: AttachedResult[];
+
+  /**
+   * A potential error message. null if success
+   */
+  message?: string;
+}
+
+export interface SetAttachToCaseAttachActionCreatorPayload {
+  result: AttachedResult;
+}
+
+export interface SetAttachToCaseDetachActionCreatorPayload {
+  result: AttachedResult;
+}
+
+const RequiredAttachedResultRecord = new RecordValue({
+  options: {
+    required: true,
+  },
+  values: {
+    articleLanguage: nonEmptyString,
+    articlePublishStatus: nonEmptyString,
+    articleVersionNumber: nonEmptyString,
+    caseId: requiredNonEmptyString,
+    knowledgeArticleId: nonEmptyString,
+    name: nonEmptyString,
+    permanentId: nonEmptyString,
+    resultUrl: nonEmptyString,
+    source: nonEmptyString,
+    title: requiredNonEmptyString,
+    uriHash: nonEmptyString,
+  },
+});
+
+export const setAttachedResults = createAction(
```

Comment:
Don't we need to validate that at least one of the two fields, `permanentId` and `uriHash `, is present in the results  we try to attach with this action?

Reply:
Because of "reasons" I don't want to "crash" if you load attached results that have bad data.

I think in the past there was even a third use-case where you could only load CaseAttach results (Native Salesforce attached knowledge articles, non-coveo results) those would have neither field.

But I am not certain of the backwards-compatibility we want to keep with that feature...

So let's say you'd load results that contain neither, it'd be okay, they just wouldn't show up as attached ever.
But IMO that shouldn't "crash" the attached results feature just because you have 1 bad result with missing field values I think anyway.

Reply:
Makes a lot of sense! thank you for the clarification👌